### PR TITLE
vimPlugins: Add catppuccin-nvim and update

### PR DESCRIPTION
**Pull request by Kranzes**
###### Motivation for this change
- Added [catppuccin-nvim](https://github.com/catppuccin/nvim)
- Updated all plugins
- Changed vim-clap's `cargoSha256`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Kranzes** [comment]
Result of `nixpkgs-review pr 156123` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>vimPlugins.vim-clap</li>
  </ul>
</details>
<details>
  <summary>35 packages built:</summary>
  <ul>
    <li>spacevim</li>
    <li>vimPlugins.Coqtail</li>
    <li>vimPlugins.SchemaStore-nvim</li>
    <li>vimPlugins.catppuccin-nvim</li>
    <li>vimPlugins.chadtree</li>
    <li>vimPlugins.coc-nvim</li>
    <li>vimPlugins.completion-treesitter</li>
    <li>vimPlugins.friendly-snippets</li>
    <li>vimPlugins.glow-nvim</li>
    <li>vimPlugins.kanagawa-nvim</li>
    <li>vimPlugins.kommentary</li>
    <li>vimPlugins.lean-nvim</li>
    <li>vimPlugins.lightspeed-nvim</li>
    <li>vimPlugins.ncm2-ultisnips</li>
    <li>vimPlugins.neorg</li>
    <li>vimPlugins.neovim-ayu</li>
    <li>vimPlugins.nord-nvim</li>
    <li>vimPlugins.nordic-nvim</li>
    <li>vimPlugins.null-ls-nvim</li>
    <li>vimPlugins.nvim-lspconfig</li>
    <li>vimPlugins.nvim-tree-lua</li>
    <li>vimPlugins.nvim-treesitter</li>
    <li>vimPlugins.nvim-treesitter-textobjects</li>
    <li>vimPlugins.nvim-ts-rainbow</li>
    <li>vimPlugins.nvim_context_vt</li>
    <li>vimPlugins.octo-nvim</li>
    <li>vimPlugins.refactoring-nvim</li>
    <li>vimPlugins.surround-nvim</li>
    <li>vimPlugins.telescope-github-nvim</li>
    <li>vimPlugins.ultisnips</li>
    <li>vimPlugins.vim-table-mode</li>
    <li>vimPlugins.vim-tpipeline</li>
    <li>vimPlugins.vim-ultest</li>
    <li>vimPlugins.vim-wakatime</li>
    <li>vimPlugins.vimspector</li>
  </ul>
</details>

**Comment by Kranzes** [comment]
Result of `nixpkgs-review pr 156123` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>36 packages built:</summary>
  <ul>
    <li>spacevim</li>
    <li>vimPlugins.Coqtail</li>
    <li>vimPlugins.SchemaStore-nvim</li>
    <li>vimPlugins.catppuccin-nvim</li>
    <li>vimPlugins.chadtree</li>
    <li>vimPlugins.coc-nvim</li>
    <li>vimPlugins.completion-treesitter</li>
    <li>vimPlugins.friendly-snippets</li>
    <li>vimPlugins.glow-nvim</li>
    <li>vimPlugins.kanagawa-nvim</li>
    <li>vimPlugins.kommentary</li>
    <li>vimPlugins.lean-nvim</li>
    <li>vimPlugins.lightspeed-nvim</li>
    <li>vimPlugins.ncm2-ultisnips</li>
    <li>vimPlugins.neorg</li>
    <li>vimPlugins.neovim-ayu</li>
    <li>vimPlugins.nord-nvim</li>
    <li>vimPlugins.nordic-nvim</li>
    <li>vimPlugins.null-ls-nvim</li>
    <li>vimPlugins.nvim-lspconfig</li>
    <li>vimPlugins.nvim-tree-lua</li>
    <li>vimPlugins.nvim-treesitter</li>
    <li>vimPlugins.nvim-treesitter-textobjects</li>
    <li>vimPlugins.nvim-ts-rainbow</li>
    <li>vimPlugins.nvim_context_vt</li>
    <li>vimPlugins.octo-nvim</li>
    <li>vimPlugins.refactoring-nvim</li>
    <li>vimPlugins.surround-nvim</li>
    <li>vimPlugins.telescope-github-nvim</li>
    <li>vimPlugins.ultisnips</li>
    <li>vimPlugins.vim-clap</li>
    <li>vimPlugins.vim-table-mode</li>
    <li>vimPlugins.vim-tpipeline</li>
    <li>vimPlugins.vim-ultest</li>
    <li>vimPlugins.vim-wakatime</li>
    <li>vimPlugins.vimspector</li>
  </ul>
</details>

File: pkgs/misc/vim-plugins/vim-plugin-names [hunk 1]
```diff
@@ -63,6 +63,7 @@ buoto/gotests-vim
 camspiers/lens.vim
 camspiers/snap
 carlitux/deoplete-ternjs
+catppuccin/nvim as catppuccin-nvim
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/vim-plugin-names:66]: Do we support this syntax? Seeing it the first time right now.

Comment (Kranzes) [reply pkgs/misc/vim-plugins/vim-plugin-names:66]: Dracula theme uses it too, since the github repo is just called nvim/vim.

**Pull request merged**